presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -551,6 +554,14 @@ private PrestoConnection connection()
         if (connection.isClosed()) {
             throw new SQLException("Connection is closed");
         }
+        return updateSessionProperties(connection);
+    }
+
+    private PrestoConnection updateSessionProperties(PrestoConnection connection)
+    {
+        if (queryTimeoutSeconds.get() > 0) {
+            connection.setSessionProperty(QUERY_MAX_RUN_TIME, queryTimeoutSeconds.longValue() + "s");
```

Comment:
PrestoConnection is shared across multiple statements, but the timeout should only apply to this statement. We might need some refactoring.